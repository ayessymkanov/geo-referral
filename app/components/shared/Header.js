import React from 'react'
import PropTypes from 'prop-types'
import { Header, HeaderLeft, HeaderCenter, HeaderRight } from './styles'

const Component = ({
  label,
  LeftButton,
  leftBtnOnClick,
  leftBtnShown,
  RightButton,
  rightBtnOnClick,
  rightBtnShown
}) => {
  return (
    <Header>
      <HeaderLeft>{leftBtnShown && <LeftButton onClick={leftBtnOnClick} />}</HeaderLeft>
      <HeaderCenter>{label}</HeaderCenter>
      <HeaderRight>{rightBtnShown && <RightButton onClick={rightBtnOnClick} />}</HeaderRight>
    </Header>
  )
}

Component.propTypes = {
  label: PropTypes.string,
  LeftButton: PropTypes.element,
  leftBtnOnClick: PropTypes.func,
  leftBtnShown: PropTypes.bool,
  RightButton: PropTypes.element,
  rightBtnOnClick: PropTypes.func,
  rightBtnShown: PropTypes.bool
}

Component.defaultProps = {
  label: '',
  LeftButton: null,
  leftBtnOnClick: () => {},
  leftBtnShown: false,
  RightButton: null,
  rightBtnOnClick: () => {},
  rightBtnShown: false
}

export default Component
