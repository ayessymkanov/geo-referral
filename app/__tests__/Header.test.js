import React from 'react'
import Header from '../components/shared/Header'
import renderer from 'react-test-renderer'

test('Header test', () => {
  const component = renderer.create(
    <Header label="Hello" />
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

})