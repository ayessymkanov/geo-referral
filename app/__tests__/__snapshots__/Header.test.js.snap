// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test 1`] = `
<div
  className="sc-gzVnrw CaTve"
>
  <div
    className="sc-htoDjs fRBlsV"
  />
  <div
    className="sc-dnqmqq hRirXI"
  >
    Hello
  </div>
  <div
    className="sc-iwsKbI bofxaI"
  />
</div>
`;
